src/virtual-machine.js

```diff
@@ -543,17 +543,57 @@ class VirtualMachine extends EventEmitter {
     /**
      * Get an SVG string from storage.
      * @param {int} costumeIndex - the index of the costume to be got.
-     * @return {string} the costume's SVG string, or null if it's not an SVG costume.
+     * @return {string|HTMLImageElement} the costume's SVG string if it's SVG,
+     *     an HTMLImageElement if it's a PNG, or null if it couldn't be found or decoded.
      */
     getCostume (costumeIndex) {
         const id = this.editingTarget.getCostumes()[costumeIndex].assetId;
-        if (id && this.runtime && this.runtime.storage &&
-                this.runtime.storage.get(id).dataFormat === 'svg') {
+        if (!id || !this.runtime || !this.runtime.storage) return null;
+        const format = this.runtime.storage.get(id).dataFormat;
+        if (format === this.runtime.storage.DataFormat.SVG) {
             return this.runtime.storage.get(id).decodeText();
+        } else if (format === this.runtime.storage.DataFormat.PNG ||
+                format === this.runtime.storage.DataFormat.JPG) {
+            const data = this.runtime.storage.get(id).encodeDataURI();
+            const image = new Image();
+            image.src = data;
+            return image;
         }
+        log.error(`Unhandled format: ${this.runtime.storage.get(id).dataFormat}`);
         return null;
     }
 
+    /**
+     * Update a costume with the given bitmap
+     * @param {int} costumeIndex - the index of the costume to be updated.
+     * @param {HTMLCanvasElement} bitmap - new bitmap for the renderer.
```

Comment:
If possible I think limiting the amount of DOM stuff in the VM would be good. Passing in an [`ImageData`](https://developer.mozilla.org/en-US/docs/Web/API/ImageData) object (or a vanilla JS object with the same properties) is how this type of data is interfaced from the renderer. 

Is it possible to have the caller do the `getImageData` and passing the result instead of the canvas element? The ImageData also includes width/height properties.

Reply:
Passed out ImageData. Couldn't get the image data to go into storage though without converting to canvas first still